Got basic GraphQL working in frontend

danielreti opened the pull request.

illusional commented on web/src/index.tsx:
```diff
@@ -1,19 +1,36 @@
 
+// 2
+const httpLink = createHttpLink({
+    uri: "http://localhost:8000/graphql",
```
Can this just be `/graphql`:

```suggestion
    uri: "/graphql",
```

danielreti replied:
Fixed

illusional commented on web/src/graphQL.tsx:
```diff
@@ -0,0 +1,23 @@
+    console.log(data.myProjects);
+    return (
+        <>
+            {data.myProjects.map((item: { id: string }) => (
```
Awesome! Do you get any type-safety here? Or would you basically have to create an interface for each endpoint?

danielreti replied:
Added :)

danielreti commented:
This became a little messy I think but it seems to be on the right track now. 

Before deploying locally, you need to run `strawberry export-schema api.graphql.schema:schema > web/src/schema.graphql` in the root directory to generate a GraphQL schema. Then running `npm run compile` before `npm start` creates the typed gql interfaces. It also creates some new files which I added to gitignore along with the schema (which should regenerate every deploy I think). We'll need to add this flow to the deploy actions too. 

Happy to discuss and make changes as obviously this is not my area and I'm not sure of the best way to do things. Also would love a double check on a few things that changed when I updated things and I'm not sure why, like the jsx compiler options in `tsconfig.json` changing from `'react'` to `'react-jsx'` and the fact that TypeScript now seems to prevent compilation when there are errors (which I like) whereas previously it would still compile and then show errors in the console (and break the page if they were breaking errors).

illusional commented:
Can you make `npm compile` also run `strawberry export` (and probably also `npm build` run the compile, which also transitively gets the schema). That way you don't have to change the github action when compiling.

illusional commented on web/src/graphQL.tsx:
```diff
@@ -0,0 +1,26 @@
+        }
+    }
+`);
+export const GraphQL: React.FunctionComponent<{}> = () => {
```
Is this a test component, or a ProjectSelector?

danielreti replied:
Just a test component. It's still a draft PR, just wanted to get something working. Once all the schema setup (+linting etc) have been setup, I'll delete this component and start making bigger changes.

illusional commented on web/src/index.tsx:
```diff
@@ -1,19 +1,36 @@
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React from "react";
```
Can you standardise quotes + formatting for the projects in one PR, this reformatting clutters the PR changes when reviewing

danielreti commented:
Problem is you can only run the `strawberry export` in the root directory not `web` as it doesn't support relative imports for some reason?? (You get this error: `TypeError: the 'package' argument is required to perform a relative import`) Otherwise I would have just added it to the `npm build` already

illusional commented:
You should be able to change directory within a compile, eg ([source](https://stackoverflow.com/a/54390944/2860731)):

```
(cd /path/to/your/app && npm start && cd -)
```
